### PR TITLE
Add VMService command to get frame rasterization metrics

### PR DESCRIPTION
**Pull request by iskakaushik**
This change lets the user type in 'j' / 'J' (short for jank) to retrieve the raster metrics for the last frame held by the rasterizer. This is typically the frame currently being seen by the user. The result is dumped to a file in the app-directory like so: `flutter_jank_metrics_01.json`.

An example of one such file is: https://gist.github.com/iskakaushik/c4117c8533828f3ed89495d3f23f8fa0
This file can be visualized as the user pleases, one such location is: https://iska.is/jank64

### CODE REVIEW

**Comment by iskakaushik** [comment]
cc: @gaaclarke

**Review by dnfield — APPROVED** [review]
LGTM.

@christopherfujino should get a chance to take a look too.

**Review by christopherfujino — COMMENTED** [review]
This is really cool!

Can you add an entry for this to https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/command_help.dart#L164, and also print it https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/resident_runner.dart#L1433

File: packages/flutter_tools/test/general.shard/vmservice_test.dart [hunk 1]
```diff
@@ -459,6 +459,14 @@ void main() {
           method: 'getVMTimeline',
           errorCode: RPCErrorCodes.kServiceDisappeared,
         ),
+        const FakeVmServiceRequest(
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/vmservice_test.dart:462]: Can you also add a successful test where the stats are logged?

**Review by christopherfujino — CHANGES_REQUESTED** [review]

File: packages/flutter_tools/test/general.shard/vmservice_test.dart (hunk 1 above)
Comment (iskakaushik) [reply packages/flutter_tools/test/general.shard/vmservice_test.dart:462]: Done!

**Review by christopherfujino — APPROVED** [review]
LGTM!

**Comment by christopherfujino** [comment]
Looks like we test the command help output in (file:///b/s/w/ir/x/w/flutter/packages/flutter_tools/test/general.shard/resident_runner_test.dart:1312:5)

**Comment by iskakaushik** [comment]
> Looks like we test the command help output in (file:///b/s/w/ir/x/w/flutter/packages/flutter_tools/test/general.shard/resident_runner_test.dart:1312:5)

yup, just noticed and fixed it :-)

**Pull request merged**